### PR TITLE
feat(styles): add new examples for ShellBar

### PR DESCRIPTION
**Pull request by InnaAtanasova**
## Related Issue
Closes SAP/fundamental-styles#

Updated styling for the search field in shellbar

### CODE REVIEW

File: packages/styles/src/shellbar.scss [hunk 1]
```diff
@@ -135,163 +196,283 @@ $block: #{$fd-namespace}-shellbar;
...
+
+        &::before {
+          position: absolute;
+          content: "";
+          outline: var(--fdShellbar_Search_Active_Outline);
+          outline-offset: -0.125rem;
+          top: 0;
+          right: 0;
+          left: 0;
+          bottom: 0;
```
Comment (InnaAtanasova) [inline packages/styles/src/shellbar.scss:400]: you can use this mixin here:
```
@mixin fd-set-equal-positions($value) {
  top: $value;
  right: $value;
  bottom: $value;
  left: $value;
}
```

File: packages/styles/src/shellbar.scss [hunk 2]
```diff
@@ -135,163 +196,283 @@ $block: #{$fd-namespace}-shellbar;
...
+          }
+
+          .fd-select__button {
+            background-color: var(--fdShellbar_Search_Category_Button_Hover_Background);
+          }
+        }
+
+        .fd-select {
+          @include fd-set-margin-right(var(--fdShellbar_Search_Category_Offset_Right));
+        }
```
Comment (InnaAtanasova) [inline packages/styles/src/shellbar.scss:309]: These 2 rules you can't use as it violates the self-contained rules. You are using as a selector a class that belongs to another component

File: packages/styles/src/shellbar.scss [hunk 3]
```diff
@@ -135,163 +196,283 @@ $block: #{$fd-namespace}-shellbar;
...
+              display: none;
+            }
+          }
+        }
+
+        .fd-select__button {
+          background: transparent;
+          background-color: var(--fdShellbar_Input_Background);
+          color: var(--sapShell_InteractiveTextColor);
+        }
```
Comment (InnaAtanasova) [inline packages/styles/src/shellbar.scss:350]: same here

File: packages/styles/stories/Components/shellbar/shellbar.stories.js [hunk 4]
```diff
@@ -126,28 +125,25 @@ export const ProductMenuAndSearch = () => `<div style="height:200px">
...
+        <div class="fd-shellbar__group fd-shellbar__group--center">
+            <div class="fd-shellbar__action">
                 <div class="fd-popover__control">
                     <div aria-label="Image label" onclick="onPopoverClick('F4GcX348b')" aria-controls="F4GcX348b" aria-expanded="false" aria-haspopup="true">
-                        <div class="fd-input-group fd-shellbar__input-group">
-                            <input aria-label="search-input" type="text" class="fd-input fd-input-group__input fd-shellbar__input-group-input" id="F4GcX348b1" placeholder="Search">
-                            <span class="fd-input-group__addon fd-shellbar__input-group-addon fd-input-group__addon--button">
-                                <button aria-label="button-decline" class="fd-shellbar__button fd-button fd-button--transparent">
-                                        <i class="sap-icon--decline"></i>
+                        <div class="fd-input-group fd-shellbar__search-field">
```
Comment (platon-rov) [inline packages/styles/stories/Components/shellbar/shellbar.stories.js:134]: this is a breaking change, isn't it?

**Review by platon-rov — CHANGES_REQUESTED** [review]
Buttons active state, color isn't blue
image

Specs
image

Buttons hover, active state, HC themes, missing border
image

Specs
image

File: packages/styles/stories/Components/shellbar/shellbar.stories.js (hunk 4 above)
Comment (N1XUS) [reply packages/styles/stories/Components/shellbar/shellbar.stories.js:134]: @platon-rov from what I've seen in ngx, it wasnt used at all, but yes, it's a breaking change

File: packages/styles/src/shellbar.scss [hunk 5]
```diff
@@ -135,163 +196,277 @@ $block: #{$fd-namespace}-shellbar;
...
+          }
+        }
 
-    &-input {
-      border: none;
-      color: var(--fdShellbar_Input_Color);
-      position: relative;
+        &:not(:placeholder-shown) {
+          & ~ .#{$block}__search-submit {
+            .fd-button {
```
Comment (InnaAtanasova) [inline packages/styles/src/shellbar.scss:249]: can't have this, unfortunately

Comment (InnaAtanasova) [reply packages/styles/src/shellbar.scss:249]: apart from that, the rest looks great!

File: packages/styles/stories/Components/shellbar/shellbar.stories.js (hunk 4 above)
Comment (platon-rov) [reply packages/styles/stories/Components/shellbar/shellbar.stories.js:134]: Please rebase and rewrite commit message to note breaking changes/

File: packages/styles/stories/Components/shellbar/shellbar.stories.js [hunk 6]
```diff
@@ -743,10 +837,130 @@ ProductSwitch.parameters = {
...
+</div>
+`;
+
+ResponsivePaddings.storyName = 'ResponsivePaddings';
+
+ResponsivePaddings.parameters = {
+    docs: {
+        description: {
+            story: `
+bla bla bla
```
Comment (platon-rov) [inline packages/styles/stories/Components/shellbar/shellbar.stories.js:962]: Please update

**Review by platon-rov — CHANGES_REQUESTED** [review]
Button border looks broken when is in active state
image

Chevron icon should be blue when button is in active state
image

Background of button should be orange on hover, HC
image

No border left on button
image

This pr https://github.com/SAP/fundamental-styles/pull/4198 may solve issues with button borders in HC themes, let's try to merge it and rebase this branch

File: packages/styles/stories/Components/shellbar/shellbar.stories.js (hunk 6 above)
Comment (N1XUS) [reply packages/styles/stories/Components/shellbar/shellbar.stories.js]: Should be fixed now

**Review by platon-rov — CHANGES_REQUESTED** [review]
Note breaking changes (thread in comments above).

"|" broken, RTL
image

Button border remains after click on the select's dropdown button
image

**Comment by N1XUS** [comment]
> Note breaking changes (thread in comments above).
> 
> "|" broken, RTL image
> 
> Button border remains after click on the select's dropdown button image

The border is there because drop-down is kinda opened (it doesn’t have a list), so that’s fine. As to border on rtl, it’s been fixed with last commit

**Review by platon-rov — APPROVED** [review]

**Pull request merged**